source/Octopus.Tentacle.Tests.Integration/TentacleCommandLineTests.cs

```diff
@@ -0,0 +1,659 @@
+    }},
+    ""Services"": {{
+      ""ListenIP"": null,
+      ""NoListen"": false,
+      ""PortNumber"": {clientAndTentacle.RunningTentacle.ServiceUri.Port}
+    }}
+  }}
+}}
+");
+            }
+
+            await Task.CompletedTask;
+        }
+
+        [Test]
+        [TentacleConfigurations(testCurrentVersionOnly: true)]
+        [WindowsTest]
+        [NonParallelizable]
+        public async Task WatchdogCreateAndDeleteCommand(TentacleConfigurationTestCase tc)
+        {
+            await using var clientAndTentacle = await tc.CreateBuilder().Build(CancellationToken);
+            var create = await RunCommandWithValidExitCodeAssert(tc, $"watchdog --create --instances={clientAndTentacle.RunningTentacle.InstanceName}");
+            create.StdError.Should().BeNullOrEmpty();
+            create.StdOut.Should().ContainEquivalentOf("Creating watchdog task");
+            var delete = await RunCommandWithValidExitCodeAssert(tc, $"watchdog --delete --instances={clientAndTentacle.RunningTentacle.InstanceName}");
+            delete.StdError.Should().BeNullOrEmpty();
+            delete.StdOut.Should().ContainEquivalentOf("Removing watchdog task");
+        }
+        
+        FileVersionInfo GetVersionInfo(TentacleConfigurationTestCase tentacleConfigurationTestCase)
+        {
+            var tentacleExe = TentacleExeFinder.FindTentacleExe(tentacleConfigurationTestCase.TentacleRuntime);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return FileVersionInfo.GetVersionInfo(tentacleExe);
+            }
+
+            //todo: change this to trust the value set in the context.TentacleExePath (will need a renovation of ExePathResolver to be non windows specific)
+            return FileVersionInfo.GetVersionInfo($"{tentacleExe}.dll");
+        }
+
+        async Task<(int ExitCode, string StdOut, string StdError)> RunCommandWithValidExitCodeAssert(TentacleConfigurationTestCase tentacleConfigurationTestCase, string input)
+        {
+            var (exitCode, stdout, stderr) = await RunCommand(tentacleConfigurationTestCase, input);
+            exitCode.Should().Be(0, $"we expected the command to succeed.\r\nStdErr: '{stderr}'\r\nStdOut: '{stdout}'");
+            return (exitCode, stdout, stderr);
+        }
+
+        async Task<(int ExitCode, string StdOut, string StdError)> RunCommand(TentacleConfigurationTestCase tentacleConfigurationTestCase, string input)
+        {
+            var tentacleExe = TentacleExeFinder.FindTentacleExe(tentacleConfigurationTestCase.TentacleRuntime);
+            var args = input;
+            var output = new StringBuilder();
+            var errorOut = new StringBuilder();
+            
+            var result = await RetryHelper.RetryAsync<CommandResult, CommandExecutionException>(
+                () => Cli.Wrap(tentacleExe)
+                    .WithArguments(args)
```

Comment:
Note the docs on this method:
```
/// <summary>
    /// Creates a copy of this command, setting the arguments to the specified value.
    /// </summary>
    /// <remarks>
    /// Avoid using this overload, as it requires the arguments to be escaped manually.
    /// Formatting errors may lead to unexpected bugs and security vulnerabilities.
    /// </remarks>
```

Reply:
Changed to use the string[] overload for passing arguments